rfc/0070-normalise-mod-ids.md

```diff
@@ -0,0 +1,49 @@
+# Normalise Mod ID Separators
+
+## Summary
+
+Treat dashes and underscores as the same for mod id comparison purposes.
+
+## Motivation
+
+An occasional cause for confusion is when two different mods only differ by underscores (`_`) and dashes(`-`) in their mod-ids. Since those characters aren't usually used to discriminate between mods, I propose we consider them as the same for the purposes of mod resolution and solving.
+
+## Explanation
+
+I propose we normalise all quilt mod-ids by replacing every dash (`-`) and underscore (`_`) with an internal character (likely `#`, but any character that's not permitted by the specification, and is unlikely to be part of the specification in the future, would work. Ascii is preferable due to string optimisations though).
+
+This normalisation would purely happen internally to quilt-loader, and wouldn't be shown to users, or exposed in methods like `ModContainer.getId()`. Instead this would allow dependencies and breaks to accidently target the wrong ID without causing issues, and would automatically cause mods with underscores or dashes to "provide" the other representations automatically.
+
+For example, a mod with an ID "potion_craft-expanded" would result in:
+- `QuiltLoader.getModContainer("potion_craft-expanded")` would return the mod
+- `QuiltLoader.getModContainer("potion-craft-expanded")` would return the same mod
+- The same mod is also returned for strings `potion_craft_expanded` and `potion-craft_expanded`
+- The returned `ModContainer` will return `potion_craft-expanded` from `ModContainer.metadata().id()`.
+- `QuiltLoader.isModLoaded` would return true for all 4 strings
+- `QuiltLoader.getEntrypoints` would return the same entrypoint list for each id.
+- `QuiltLoader.getAllMods()` will only contain a single entry
+- `ModMetadata.provides()` would NOT contain provided entries for every possible combination, instead it would be empty (if no provided entries exist in the quilt.mod.json)
```

Comment:
```suggestion
- `ModMetadata.provides()` would NOT contain provided entries for every possible combination, instead it would be empty (unless the mod defines `provided` entries in its quilt.mod.json)
```